lib/matobj.gi

```diff
@@ -1419,8 +1419,32 @@ InstallMethod( ViewString,
     ViewStringForMatrixObj );
 
 InstallMethod( DisplayString,
-    [ IsMatrixOrMatrixObj ],
-    ViewStringForMatrixObj );
+[ IsMatrixOrMatrixObj ],
+function(M)
+local i,j,m,numberCols,numberRows,baseDom,zeroEle,outputstring;
+    numberRows := NrRows(M);
+    numberCols := NrCols(M);
+    baseDom := BaseDomain(M);
+    zeroEle := Zero(baseDom);
+    outputstring := "<";
+    if not IsMutable(M) then Append(outputstring, "immutable "); fi;
+    outputstring := Concatenation(outputstring,String(numberRows),"x",String(numberCols),"-matrix over ",String(baseDom),":\n");
+    Append(outputstring,"[");
```

Comment:
This causes the first line to be indented differently. I would either add a corresponding whitespace before each other line; or just make the `[` belong the end of the previous line (probably nicer). So instead of
```
<a 2x2 matrix over Rationals
[[1 2 ]
[3 4 ]
]>
```
print
```
<a 2x2 matrix over Rationals [
[ 1 2 ]
[3  4 ]
]>
```

The `ViewString` could be similar but perhaps without the entries being shown? I.e.
```
```